File: ai-service/test_aramex_pickup.py
#!/usr/bin/env python3
"""
اختبار وظيفة إنشاء طلب الاستلام في أرامكس
"""

import sys
import os

# إضافة مجلد mararsil-main إلى path
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

def test_aramex_pickup_data():
    """اختبار دالة إنشاء بيانات طلب الاستلام"""
    try:
        # استيراد الدالة
        from services.AramexService import createPickupRequestData

        # بيانات المرسل التجريبية
        shipperData = {
            "full_name": "أحمد محمد",
            "mobile": "0512345678",
            "email": "ahmed@example.com",
            "address": "شارع الملك فيصل، الرياض",
            "company_name": "متجر أحمد"
        }

        # معلومات الشحنة التجريبية
        shipmentInfo = {
            "trackingNumber": "123456789"
        }

        # إنشاء بيانات طلب الاستلام
        pickupData = createPickupRequestData(shipperData, shipmentInfo)

        print("✅ تم إنشاء بيانات طلب الاستلام بنجاح:")
        print(f"  - اسم الاتصال: {pickupData['contactName']}")
        print(f"  - رقم الهاتف: {pickupData['phone']}")
        print(f"  - البريد الإلكتروني: {pickupData['email']}")
        print(f"  - رقم التتبع: {pickupData['reference']}")
        print(f"  - وقت الاستلام: {pickupData['pickupDateTime']}")
        print(f"  - وقت الانتهاء: {pickupData['closingDateTime']}")

        return True

    except Exception as e:
        print(f"❌ خطأ في اختبار بيانات طلب الاستلام: {e}")
        import traceback
        traceback.print_exc()
        return False

def test_format_address():
    """اختبار دالة تنسيق العنوان"""
    try:
        from services.AramexService import formatAddress

        address = {
            "address": "شارع الملك فيصل، الرياض",
            "city": "الرياض",
            "country": "SA"
        }

        formatted = formatAddress(address)

        print("✅ تم تنسيق العنوان بنجاح:")
        print(f"  - Line1: {formatted['Line1']}")
        print(f"  - City: {formatted['City']}")
        print(f"  - CountryCode: {formatted['CountryCode']}")

        return True

    except Exception as e:
        print(f"❌ خطأ في اختبار تنسيق العنوان: {e}")
        import traceback
        traceback.print_exc()
        return False

def main():
    """الدالة الرئيسية للاختبار"""
    print("🧪 اختبار وظائف أرامكس - إنشاء طلب الاستلام")
    print("=" * 50)

    tests = [
        ("اختبار إنشاء بيانات طلب الاستلام", test_aramex_pickup_data),
        ("اختبار تنسيق العنوان", test_format_address),
    ]

    passed = 0
    total = len(tests)

    for test_name, test_func in tests:
        print(f"\n🔍 {test_name}:")
        print("-" * 30)

        if test_func():
            passed += 1
            print(f"✅ {test_name}: نجح")
        else:
            print(f"❌ {test_name}: فشل")

    print(f"\n📊 نتائج الاختبار: {passed}/{total} نجح")
    print("🎉 انتهى الاختبار!" if passed == total else "⚠️ يوجد أخطاء تحتاج إصلاح")

if __name__ == "__main__":
    main()

File: test_aramex_pickup.js
#!/usr/bin/env node

/**
 * اختبار شامل لميزة إنشاء طلب الاستلام في أرامكس
 */

const AramexService = require('./services/AramexService');

async function testDataFormatting() {
  console.log('🔍 اختبار تنسيق البيانات...');

  // بيانات المرسل التجريبية
  const shipperAddress = {
    full_name: "أحمد محمد التاجر",
    mobile: "0512345678",
    email: "ahmed@store.com",
    address: "شارع الملك فيصل، الرياض",
    city: "الرياض",
    country: "SA"
  };

  // معلومات الشحنة التجريبية
  const shipmentInfo = {
    trackingNumber: "123456789"
  };

  try {
    // اختبار createPickupRequestData
    const pickupData = AramexService.createPickupRequestData(shipperAddress, shipmentInfo);

    console.log('✅ تم إنشاء بيانات طلب الاستلام:');
    console.log('  - pickupAddress:', JSON.stringify(pickupData.pickupAddress, null, 2));
    console.log('  - contactName:', pickupData.contactName);
    console.log('  - pickupDateTime:', pickupData.pickupDateTime);
    console.log('  - reference:', pickupData.reference);

    return pickupData;
  } catch (error) {
    console.error('❌ خطأ في تنسيق البيانات:', error.message);
    throw error;
  }
}

async function testAramexPickup(pickupData) {
  console.log('\n🚛 اختبار إنشاء طلب الاستلام...');

  try {
    const result = await AramexService.createPickupRequest(
      pickupData.shipperAddress || {
        full_name: "أحمد محمد التاجر",
        mobile: "0512345678",
        email: "ahmed@store.com",
        address: "شارع الملك فيصل، الرياض",
        city: "الرياض",
        country: "SA"
      },
      { trackingNumber: "123456789" }
    );

    console.log('📊 نتيجة الاختبار:');
    console.log('  - نجح:', result.success);
    console.log('  - الرسالة:', result.message);

    if (result.success) {
      console.log('  - رقم طلب الاستلام:', result.pickupId);
      console.log('  - التاريخ المجدول:', result.scheduledDate);
      console.log('✅ تم إنشاء طلب الاستلام بنجاح!');
    } else {
      console.log('  - الخطأ:', result.error);
      console.log('⚠️ فشل في إنشاء طلب الاستلام (قد يكون بسبب بيانات الاختبار)');
    }

    return result;
  } catch (error) {
    console.error('❌ خطأ في الاختبار:', error.message);
    console.error('Stack:', error.stack);
    throw error;
  }
}

async function main() {
  console.log('🧪 اختبار شامل لميزة إنشاء طلب الاستلام في أرامكس');
  console.log('=' .repeat(60));

  try {
    // اختبار 1: تنسيق البيانات
    const pickupData = await testDataFormatting();

    // اختبار 2: إنشاء طلب الاستلام
    const result = await testAramexPickup(pickupData);

    console.log('\n🎉 انتهى الاختبار بنجاح!');

  } catch (error) {
    console.error('\n❌ فشل الاختبار:', error.message);
    process.exit(1);
  }
}

// تشغيل الاختبار
main().catch(console.error);
